lua/gitlinker/highlight.lua

```diff
@@ -0,0 +1,36 @@
+local M = {}
+
+-- Highlights the text selected by the specified range.
+--- @param range Range?
+M.show = function(range)
+    if not range then
+        return
+    end
+    local namespace = vim.api.nvim_create_namespace("NvimGitLinker")
+    local lstart, lend = range.lstart, range.lend
+    if lend and lend < lstart then
+        lstart, lend = lend, lstart
+    end
+    local pos1 = { lstart - 1, 1 }
+    local pos2 = { (lend or lstart) - 1, vim.fn.col("$") }
+    vim.highlight.range(
+        0,
+        namespace,
+        "NvimGitLinkerHighlightTextObject",
+        pos1,
+        pos2,
+        { inclusive = true }
+    )
+    -- Force the screen to highlight the text immediately
+    vim.cmd("redraw")
+end
+
+-- Clears the gitlinker highlights for the buffer.
+M.clear = function()
+    local namespace = vim.api.nvim_create_namespace("NvimGitLinker")
```

Comment:
hi @stevanmilic , would you please educate me, what does `nvim_create_namespace` API doing here? it looks related to the highlighting syntax  `NvimGitLinkerHighlightTextObject`.

Even I'm reading the document here: https://neovim.io/doc/user/api.html#nvim_create_namespace(), I'm still not sure what it's doing, since it's the first time I'm looking at this API,